### PR TITLE
SLI-28 Analyze entire project

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeAllFilesAction.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    @Override
+    public void onSuccess() {
+      Map<VirtualFile, Collection<LiveIssue>> map = affectedFiles.stream()
+        .collect(Collectors.toMap(Function.identity(), issueManager::getForFile));
+      allFilesIssues.set(map);
+      showAllFilesTab();
+    }
+
+    private void showAllFilesTab() {
+      UIUtil.invokeLaterIfNeeded(() ->ServiceManager.getService(project, IssuesViewTabOpener.class).open(SonarLintToolWindowFactory.TAB_ALL_FILES));
```
Comment (henryju) [inline src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeAllFilesAction.java]: Code style issue

File: src/main/java/org/sonarlint/intellij/messages/AllFilesIssuesListener.java [hunk 2]
```diff
@@ -17,24 +17,19 @@
...
-import com.intellij.ui.content.ContentManager;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.util.messages.Topic;
+import java.util.Collection;
+import java.util.Map;
+import org.sonarlint.intellij.issue.LiveIssue;
 
-public class ChangedFilesTabOpener implements Runnable {
-  private final ToolWindow toolWindow;
+public interface AllFilesIssuesListener {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/messages/AllFilesIssuesListener.java]: CRITICAL Move constants to a class or enum. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1214)
CRITICAL Annotate the "AllFilesIssuesListener" interface with the @FunctionalInterface annotation [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1609)

File: src/main/java/org/sonarlint/intellij/ui/IssuesViewTabOpener.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    ApplicationManager.getApplication().assertIsDispatchThread();
+
+    ToolWindowManager toolWindowManager = ToolWindowManager.getInstance(project);
+    ToolWindow toolWindow = toolWindowManager.getToolWindow(SonarLintToolWindowFactory.TOOL_WINDOW_ID);
+    if (toolWindow != null) {
+      toolWindow.show(new ContentSelector(toolWindow, tab));
+    }
+  }
+
+  private class ContentSelector implements Runnable {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/IssuesViewTabOpener.java]: MAJOR Make this a "static" inner class. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694)

File: src/main/java/org/sonarlint/intellij/ui/SonarLintAllFilesPanel.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import com.intellij.util.messages.MessageBusConnection;
+import com.intellij.util.ui.tree.TreeUtil;
+import java.awt.BorderLayout;
+import javax.swing.JPanel;
+import org.sonarlint.intellij.core.ProjectBindingManager;
+import org.sonarlint.intellij.issue.AllFilesIssues;
+import org.sonarlint.intellij.messages.AllFilesIssuesListener;
+import org.sonarlint.intellij.messages.StatusListener;
+
+public class SonarLintAllFilesPanel extends AbstractIssuesPanel implements OccurenceNavigator, DataProvider {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintAllFilesPanel.java]: MAJOR This class has 7 parents which is greater than 5 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
MINOR "OccurenceNavigator" is implemented by a super class; there is no need to implement it here. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1939)
MINOR "DataProvider" is implemented by a super class; there is no need to implement it here. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1939)

File: src/main/java/org/sonarlint/intellij/ui/SonarLintAllFilesPanel.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  private static final String GROUP_ID = "SonarLint.alltoolwindow";
+  private static final String SPLIT_PROPORTION_PROPERTY = "SONARLINT_RESULTS_SPLIT_PROPORTION";
+
+  private final LastAnalysisPanel lastAnalysisPanel;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintAllFilesPanel.java]: CRITICAL Make "lastAnalysisPanel" transient or serializable. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1948)

File: src/main/java/org/sonarlint/intellij/ui/SonarLintAllFilesPanel.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  private final AllFilesIssues allFilesIssues;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintAllFilesPanel.java]: CRITICAL Make "allFilesIssues" transient or serializable. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1948)

File: src/main/java/org/sonarlint/intellij/ui/SonarLintToolWindowFactory.java [hunk 7]
```diff
@@ -84,6 +87,18 @@ private void addChangedFilesTab(Project project, ToolWindow toolWindow) {
     }
   }
 
+  private void addAllFilesTab(Project project, ToolWindow toolWindow) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintToolWindowFactory.java]: MAJOR Make "addAllFilesTab" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeAllFilesAction.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+      return;
+    }
+
+    SonarLintSubmitter submitter = SonarLintUtils.get(project, SonarLintSubmitter.class);
+    Collection<VirtualFile> allFiles = getAllFiles(project);
+    AnalysisCallback callback = new ShowIssuesCallable(project, allFiles);
+    submitter.submitFiles(allFiles, TriggerType.ACTION, callback, false);
+  }
+
+  private Collection<VirtualFile> getAllFiles(Project project) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeAllFilesAction.java]: MAJOR Make "getAllFiles" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 11 issues
* ![CRITICAL][CRITICAL] 4 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SonarLintCheckinHandler.java#L61](https://github.com/SonarSource/sonarlint-intellij/blob/58c6140a57be7aabe7e61451acb96d9c0df1219f/src/main/java/org/sonarlint/intellij/trigger/SonarLintCheckinHandler.java#L61): Remove this unused "toolWindowManager" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request merged**